src/core/Core.js

```diff
@@ -415,7 +415,7 @@ class Uppy {
       throttledCalculateProgress(data)
     })
 
-    this.on('core:upload-success', (fileID, uploadResp, uploadURL) => {
```

Comment:
> Can’t we just return whatever the server returns

Like the XHR object? I think the parsed JSON object it returned previously was nice.

We could also instead of a `getUploadUrl` option have a `getUploadResponse` option, and have that do `JSON.parse(xhr.responseText)` by default. The S3 plugin could override it with some XML parsing.